### ISSUE TITLE
associations not returned in history pages

### ISSUE THREAD

**Issue by ginold** [c2corg/v6_api#337]
I'm not sure it is an API problem, but here it is:
whenever i try to browse an old version of an outing, I receive an error  `(...) KeyError: 'associations' ` even though the associations were present at that time.

There is no issue with waypoints or routes.

**Comment by ginold** [comment]
same for `title_prefix` in routes page -> not available.

see https://github.com/c2corg/v6_ui/issues/408

**Comment by tsauerwein** [comment]
Yes, the associations are not returned for history pages. Like in v5, e.g.: http://www.camptocamp.org/summits/37620/fr/5

Same for the title prefix, e.g.: http://www.camptocamp.org/routes/194206/fr/2

**Comment by ginold** [comment]
Ok, I don't know if the associations should or should not be returned for history pages, however @asaunier seems concerned about not having the `title_prefix`. But very likely the `title_prefix` has to do with the associations as it is the `main_waypoint's` (associated wp) title...

So, is there something to do about it?

**Comment by ginold** [comment]
and there is this problem: 

> I still have this pb: In "Comparaison des documents" page (see outings/diff/736972/fr/1164793/1164795)
When I click on "Version du....", I get an "Internal Server Error".
See http://c2corgv6.demo-camptocamp.com/outings/diff/736956/fr/1164743/1164744

**Comment by tsauerwein** [comment]
> So, is there something to do about it?

For the associations: Would you expect the associations the document had at that version? Or the associations the document has now? I tend to say this is an "extra" task (same for the title prefix).

**Comment by tsauerwein** [comment]
> When I click on "Version du....", I get an "Internal Server Error".

The API returns the document with no error:
http://c2corgv6-api.demo-camptocamp.com/outings/736956/fr/1164743

This must be a problem on the UI server side.

**Comment by asaunier** [comment]
> Yes, the associations are not returned for history pages. Like in v5, e.g.: http://www.camptocamp.org/summits/37620/fr/5

That makes sense because there is no relationship between the versions of a document and the associations at the time of those versions: adding or removing an association doesn't create a new document version.

> Same for the title prefix, e.g.: http://www.camptocamp.org/routes/194206/fr/2

Perhaps but v5 is not flawless!
It's definitely a default not to show the main waypoint name in archived versions. Because it is part of the route name: "Mont Blanc : Face N" is the real name, not "Face N".
In v5 the summit name was figured out using the most elevated associated summit.
In v6 the main waypoint id and name are part of the route attributes, aren't they? Then we should use them in the archived doc versions.

**Comment by tsauerwein** [comment]
In v6 the field `title_prefix` caches the title of the associated main waypoint, but only on the latest locale. For history pages it would be possible to retrieve the title, but I consider this as "extra".
